gensim/test/test_ensemblelda.py

```diff
@@ -0,0 +1,419 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Author: Tobias B <proxima@sezanzeb.de>
+
+"""
+Automated tests for checking the EnsembleLda Class
+"""
+
+import os
+import logging
+import unittest
+
+import numpy as np
+from copy import deepcopy
+
+from gensim.models import EnsembleLda, LdaMulticore, LdaModel
+from gensim.test.utils import datapath, get_tmpfile, common_corpus, common_dictionary
+
+num_topics = 2
+num_models = 4
+passes = 50
+
+# windows tests fail due to the required assertion precision being too high
+rtol = 1e-04 if os.name == 'nt' else 1e-05
+
+
+class TestModel(unittest.TestCase):
+    def setUp(self):
+        # same configuration for each model to make sure
+        # the topics are equal
+        random_state = 0
+
+        self.eLDA = EnsembleLda(
+            corpus=common_corpus, id2word=common_dictionary, num_topics=num_topics,
+            passes=passes, num_models=num_models, random_state=random_state,
+            topic_model_class=LdaModel,
+        )
+
+        self.eLDA_mu = EnsembleLda(
+            corpus=common_corpus, id2word=common_dictionary, num_topics=num_topics,
+            passes=passes, num_models=num_models, random_state=random_state,
+            memory_friendly_ttda=False, topic_model_class=LdaModel,
+        )
+
+    def check_ttda(self, ensemble):
+        """tests the integrity of the ttda of any ensemble"""
+        self.assertGreater(len(ensemble.ttda), 0)
+        a = ensemble.ttda.sum(axis=1)
+        b = np.ones(len(ensemble.ttda)).astype(np.float32)
+        np.testing.assert_allclose(a, b, rtol=1e-04)
+
+    def test_eLDA(self):
+        # given that the random_state doesn't change, it should
+        # always be 2 detected topics in this setup.
+        self.assertEqual(self.eLDA.stable_topics.shape[1], len(common_dictionary))
+        self.assertEqual(len(self.eLDA.ttda), num_models * num_topics)
+        self.check_ttda(self.eLDA)
+
+        # reclustering shouldn't change anything without
+        # added models or different parameters
+        self.eLDA.recluster()
+
+        self.assertEqual(self.eLDA.stable_topics.shape[1], len(common_dictionary))
```

Comment:
Please change these to pytest-style assertions.

```suggestion
        assert self.eLDA.stable_topics.shape[1] == len(common_dictionary)
```

Reply:
I changed most stuff over to pytest-style and removed the camel cased variable names and made a first pass of improving naming / readability.  One exception is that the class still inherits from `unittest.TestCase` and overrides `setUp(self)`.